pyccel/ast/operators.py

```diff
@@ -1070,7 +1070,93 @@ class PyccelIsNot(PyccelIs):
     def __repr__(self):
         return '{} is not {}'.format(self.args[0], self.args[1])
 
+
 #==============================================================================
 
+class IfTernaryOperator(PyccelOperator):
+    """Represent a ternary conditional operator in the code, of the form (a if cond else b)
+
+    Parameters
+    ----------
+    args :
+        args : type list
+        format : condition , value_if_true, value_if_false
+
+    Examples
+    --------
+    >>> from sympy import Symbol
+    >>> from pyccel.ast.core import Assign
+	>>>	from pyccel.ast.operators import IfTernaryOperator
+    >>> n = Symbol('n')
+    >>> x = 5 if n > 1 else 2
+    >>> IfTernaryOperator(PyccelGt(n > 1),  5,  2)
+    IfTernaryOperator(PyccelGt(n > 1),  5,  2)
+    """
+    _precedence = 3
+
+    def __init__(self, cond, value_true, value_false):
+        super().__init__(cond, value_true, value_false)
+
+        if self.stage == 'syntactic':
+            return
+        if isinstance(value_true , Nil) or isinstance(value_false, Nil):
+            errors.report('None is not implemented for Ternary Operator', severity='fatal')
+        if isinstance(value_true , NativeString) or isinstance(value_false, NativeString):
+            errors.report('String is not implemented for Ternary Operator', severity='fatal')
+        assert value_true.dtype in NativeNumeric
+        assert value_false.dtype in NativeNumeric
+        if value_true.dtype not in NativeNumeric:
+            assert(value_true.dtype == value_false.dtype)
+        if value_false.rank != value_true.rank :
+            errors.report('Ternary Operator results should have the same rank', severity='fatal')
+        if value_false.shape != value_true.shape :
+            errors.report('Ternary Operator results should have the same shape', severity='fatal')
+
+    def _set_dtype(self):
+        """
+        Sets the dtype and precision for IfTernaryOperator
+        """
+        if self in NativeNumeric:
+            self._dtype = max([self.value_true.dtype, self.value_false.dtype], key = NativeNumeric.index)
+        elif isinstance(self, Nil):
+            self._dtype = Nil()
```

Comment:
The dtype must always inherit from `DataType` or be `None`. `Nil` is the pyccel version of none in the code, but it is not the same as none